app/controllers/exam_templates_controller.rb

```diff
@@ -82,12 +82,21 @@ def generate
     index = params[:examTemplateIndex].to_i
     assignment = Assignment.find(params[:assignment_id])
     exam_template = assignment.exam_templates.find(params[:id])
-    exam_template.generate_copies(copies, index)
-    flash_message(:success, t('exam_templates.generate.success', copies: copies))
 
+    flash_message(:success, t('exam_templates.generate.generate_job_started',
+                              exam_name: exam_template.assignment.short_identifier))
     generated_filename = "#{index}-#{index + copies - 1}.pdf"
-    send_file("#{exam_template.base_path}/#{generated_filename}",
-              filename: "#{generated_filename}",
+    current_job = exam_template.generate_copies(copies, index)
+    respond_to do |format|
+      format.js { render 'exam_templates/_poll_generate_job.js.erb',
+                         locals: { file_path: "#{exam_template.base_path}/#{generated_filename}",
+                                   job_id: current_job.job_id} }
+    end
+  end
+
+  def download_generate
+    send_file(params[:file_path],
+              filename: File.basename(params[:file_path]),
```

Comment:
This isn't great because it gives arbitrary access to the server's files. Rather than specifying the full path, just send back the key elements -- exam template id and generated file name -- and turn those into query parameters for the URL, and do a lookup here to compute the correct URL.